logging/journald_core.go

```diff
@@ -124,3 +132,50 @@ func encodeJournaldFieldKey(key string) string {
 
 	return key
 }
+
+// visibleFieldsMsg creates a string to be appended to the log message including fields to be explicitly printed.
+//
+// When logging against journald, the zapcore.Fields are used as journald fields, resulting in not being shown in the
+// default journalctl output (short). While this is documented in our docs, missing error messages are usually confusing
+// for end users.
+//
+// This method takes an allow list (set, map of keys to empty struct) of key to be displayed - there is the global
+// variable journaldVisibleFields; parameter for testing - and a slice of zapcore.Fields, creating an output string of
+// the allowed fields prefixed by a whitespace separator. If there are no fields to be logged, the returned string is
+// empty. So the function output can be appended to the output message without further checks.
+func visibleFieldsMsg(visibleFieldKeys map[string]struct{}, fields []zapcore.Field) string {
+	if visibleFieldKeys == nil || fields == nil {
+		return ""
+	}
+
+	enc := zapcore.NewMapObjectEncoder()
+
+	for _, field := range fields {
+		if _, shouldLog := visibleFieldKeys[field.Key]; shouldLog {
+			field.AddTo(enc)
+		}
+	}
+
+	// The internal zapcore.encodeError function[^0] can result in multiple fields. For example, an error type
+	// implementing fmt.Formatter results in another "errorVerbose" field, containing the stack trace if the error was
+	// created by github.com/pkg/errors including a stack[^1].
+	//
+	// Therefore, the keys are checked again in the following loop before being inserted in the final array, having a
+	// capacity of twice the fields length to minimize the chance of a resize.
+	//
+	// [^0]: https://github.com/uber-go/zap/blob/v1.27.0/zapcore/error.go#L47
+	// [^1]: https://pkg.go.dev/github.com/pkg/errors@v0.9.1#WithStack
+	visibleFields := make([]string, 0, 2*len(enc.Fields))
+	for k, v := range utils.IterateOrderedMap(enc.Fields) {
+		if _, shouldLog := visibleFieldKeys[k]; !shouldLog {
+			continue
+		}
+		visibleFields = append(visibleFields, fmt.Sprintf("%s=%q", k, v))
```

Comment:
Using the `%q` as a format specifier here will only work for string values, for all other types like for example `int` will result into such weird output:
```bash
2025/05/12 09:22:04 Starting Icinga DB daemon (1.3.0)
2025/05/12 09:22:04 Connecting to database at 'mysql://icingadb@localhost:3306/icingadb'
2025/05/12 09:22:04 database: Can't connect to database. Retrying	error="dial tcp [::1]:3306: connect: connection refused", int='d'
```

Reply:
Good catch. So far, I have only though about strings and errors, which are kinda strings under the hood. I have added a bit of generalization to distinct between `%q`-aware types and others. However, I would not make this too complicated as for the moment only `error` fields are relevant. Please take another look.

Reply:
Do all the values have to be quoted with `"..."`? Otherwise, why not just append the values using the `%v` verb in the default case?

Reply:
Of course they don't have to be quoted. However, I found it easier to read if each value is encapsulated, especially if multiple fields are there.

Reply:
Fine!